pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/window/aggregate/AggregateWindowFunction.java

```diff
@@ -49,54 +50,198 @@ public AggregateWindowFunction(RexExpression.FunctionCall aggCall, DataSchema in
 
   @Override
   public final List<Object> processRows(List<Object[]> rows) {
-    if (_partitionByOnly) {
-      return processPartitionOnlyRows(rows);
```

Comment:
I'd initially removed this optimization to reduce clutter since there are a lot of different cases being handled in the new function implementation. However, on second thoughts, the optimization to avoid key computation (among other things) for each row might be significant enough to be worth retaining?

Reply:
Also, the optimization is still applied to windows without `ORDER BY`, since Calcite forces the window frame to be `RANGE BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING` for such windows (and we do avoid the per row key computation for `RANGE/ROWS BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING`). So the only other case is where the partition keys and order by keys are identical.